Banking/src/main/java/com/example/Banking/service/AccountDaoImpl.java

```diff
@@ -0,0 +1,67 @@
+package com.example.Banking.service;
+import com.example.Banking.exception.IdNotFound;
+import com.example.Banking.model.Account;
+import com.example.Banking.repository.AccountRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+
+@Component
+public class AccountDaoImpl implements AccountDao {
+
+    @Autowired
+    private AccountRepository accountRepository;
+
+    //To get all accounts
+    @Override
+    public List<Account> getAll() {
+        return accountRepository.findAll();
+    }
+
+    //To get account details using id
+    public Account getAccount(long id) throws IdNotFound {
+
+            return accountRepository.findById(id).orElseThrow(() -> new IdNotFound("Account is not Found for this id"));
+
+    }
+
+    //To insert account
+    @Override
+    public void insertAccount(Account account) {
+        accountRepository.save(account);
+
+    }
+
+    //To delete account
+    @Override
+    public void deleteAccount(long id) throws IdNotFound{
+        if(accountRepository.existsById(id)) {
+            accountRepository.deleteById(id);
+        }
+        else
+        {
+            throw new IdNotFound("Account is not Found for this id");
```

Comment:
Exception should also contain the id which is causing the exception. Also do not hardcode the error strings. Create constants for error strings which can be reusable. For example delete and update can have same error strings.

Reply:
I have changed and pushed the code